src/main/java/uk/gov/hmcts/reform/idam/notify/client/NotifyService.java

```diff
@@ -41,16 +48,32 @@ public void sendSMSMessage(String from, String to, String subject, String messag
      * @should send email
      * @should do nothing if `to` param is null
      * @should throw exception if sending email throws exception
+     * @should use fallback template id if subject is default value
+     * @should use subject as template id
      */
     @Override
     public void sendEmail(String from, String to, String subject, String message, String code, Map options)
             throws AuthLoginException {
 
+        final String languageSpecificNotificationClientTemplateId;
+
+        LOGGER.fine(String.format("Sending OTP to: %s, subject: %s", to, subject));
+
+        // We are hijacking the i18n mechanism here. The language-specific templateId is being passed as message subject.
+        // This condition makes it backwards-compatible.
+        if (OPENAM_DEFAULT_SUBJECT.equalsIgnoreCase(subject)) {
```

Comment:
if we are going to be passing in a template, why not always pass in a template and not have to do this comparison of a magic string

Reply:
or is this specifically for backward compatibility

Reply:
Yes, this is specifically for backwards compatibility, in case the client gets deployed, but `idam-packer` changes not (yet). I wanted it to work with or without the HOTP library patch.